File: src/main/java/org/activecraft/activecraftcore/manager/MuteManager.java
package org.activecraft.activecraftcore.manager;

import org.activecraft.activecraftcore.events.PlayerMuteEvent;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.activecraft.activecraftcore.playermanagement.Profilev2;
import org.bukkit.Bukkit;

public class MuteManager {

    public static void mutePlayer(Profilev2 profile) {
        //call event
        PlayerMuteEvent event = new PlayerMuteEvent(profile, false);
        Bukkit.getPluginManager().callEvent(event);
        if (event.isCancelled()) return;

        profile.setTimesMuted(profile.getTimesMuted() + 1);
        profile.setMuted(true);
    }
    public static void unmutePlayer(Profilev2 profile) {
        //call event
        PlayerMuteEvent event = new PlayerMuteEvent(profile, true);
        Bukkit.getPluginManager().callEvent(event);
        if (event.isCancelled()) return;

        profile.setMuted(false);
    }
}

File: src/main/java/org/activecraft/activecraftcore/manager/MuteManager.kt
package org.activecraft.activecraftcore.manager

import org.activecraft.activecraftcore.events.PlayerMuteEvent
import org.activecraft.activecraftcore.playermanagement.Profile
import org.bukkit.Bukkit

object MuteManager {
    fun mutePlayer(profile: Profile) {
        val event = PlayerMuteEvent(profile, false)
        Bukkit.getPluginManager().callEvent(event)
        if (event.cancelled) return
        profile.timesMuted = profile.timesMuted + 1
        profile.isMuted = true
    }

    fun unmutePlayer(profile: Profile) {
        val event = PlayerMuteEvent(profile, true)
        Bukkit.getPluginManager().callEvent(event)
        if (event.cancelled) return
        profile.isMuted = false
    }
}